plugins/Goals/Controller.php

```diff
@@ -219,6 +223,25 @@ public function getEvolutionGraph(array $columns = array(), $idGoal = false, arr
         return $this->renderView($view);
     }
 
+    public function getSparklines()
+    {
+        $content = "";
+        $goals = Request::processRequest('Goals.getGoals', ['idSite' => $this->idSite, 'filter_limit' => '-1'], []);
+
+        foreach ($goals as $goal) {
+            //this is used to hack query params
+            $_GET['idGoal'] = $goal['idgoal'];
+            $_GET['allow_multiple'] = (int)$goal['allow_multiple'];
+            $_GET['only_summary'] = 1;
+            //load Visualisations Sparkline
+            $view = ViewDataTableFactory::build(Sparklines::ID, 'Goals.getMetrics', 'Goals.' . __METHOD__, true);
+            $view->config->show_title = true;
+            $content .= $view->render();
```

Comment:
Here you could use `Context::executeWithQueryParameters`, which will ensure the request parameters are reset afterwards.
```suggestion
            $params = [
                'idGoal' => $goal['idgoal'],
                'allow_multiple' => (int) $goal['allow_multiple'],
                'only_summary' => 1,
            ];

            \Piwik\Context::executeWithQueryParameters($params, function() use (&$content) {
                //load Visualisations Sparkline
                $view = ViewDataTableFactory::build(Sparklines::ID, 'Goals.getMetrics', 'Goals.' . __METHOD__, true);
                $view->config->show_title = true;
                $content .= $view->render();
            });
```